windows/src/engine/keyman32/appint/appint.h

```diff
@@ -74,28 +74,97 @@ class AppActionQueue
 	AppActionQueue();
 	virtual void ResetQueue();
 	virtual BOOL QueueAction(int ItemType, DWORD dwData);
-  
+
   BOOL IsQueueEmpty() { return QueueSize == 0; }
   int GetQueueSize() { return QueueSize; }
 };
 
 class AppContext
 {
 private:
-	WCHAR CurContext[MAXCONTEXT];
+	WCHAR CurContext[MAXCONTEXT]; //!< CurContext[0] is furthest from the caret and is null terminated.
 	int pos;
 
 public:
 	AppContext();
+  /**
+   * @brief Copy "source" AppContext to this AppContext
+   *
+   * @param source AppContext to copy
+   */
   void CopyFrom(AppContext *source);
-	void Add(WCHAR ch);
+
+  /**
+   * @brief  Add a single code unit to the Current Context. Not necessarily a complete code point
+   *
+   * @param Code unit to add
+   */
+  void Add(WCHAR ch);
+
+  /**
+   * @brief Removes a single code point from the end of the CurContext closest to the caret
+   *  - ie it will be both code units if a surrogate pair. If it is a deadkey it will remove
+   * the deadkey code point and the sentinal code point.
+   */
 	void Delete();
-	void Reset();
+
+  /**
+   * @brief Clears the CurContext and resets the position - pos - index
+   *
+   */
+  void Reset();
+
+  /**
+   * @brief Copies the characters in CurContext to supplied buffer.
+   * If bufsize is reached before the entire context was copied, the buf
+   * will be truncated to number of valid characters possible with null character
+   * termination. e.g. it will be one code unit less than buffsize if that would
+   * have meant spliting a surrogate pair
+   * @param buf  The data buffer to copy current context
+   * @param bufsize the number of code units ie size of the WCHAR buffer - not the code points
+   */
 	void Get(WCHAR *buf, int bufsize);
+
+  /**
+   * @brief Sets the CurContext to the supplied buf character array and updates the pos index.
+   *
+   * @param buf
+   */
 	void Set(const WCHAR *buf);
+
+  /**
+   * @brief Returns a pointer to the character in the current context buffer which
+   * will have at most n valid characters remaining until the the null terminating character.
+   * eg it will be one code unit less than buffsize if that would
+   * have meant spliting a surrogate pair
+   *
+   * @param n The maximum number of valid characters - (code points) not WCHAR size (code units)
+   * @return WCHAR* Pointer to the start postion for a buffer of maximum n characters
```

Comment:
```suggestion
   * @brief  Returns a pointer to the character in the current context buffer which
   *         will have at most n valid xstring units remaining until the null terminating
   *         character. It will be one code unit less than bufsize if that would
   *         have meant splitting a surrogate pair or deadkey.
   *
   * @param n        The maximum number of valid xstring units (not code points or code units)
   * @return WCHAR*  Pointer to the start postion for a buffer of maximum n xstring units
```